jni_gen/lib/src/util/name_utils.dart

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:jni_gen/src/elements/elements.dart';
+
+String getPackageName(String binaryName) => cutFromLast(binaryName, '.')[0];
+
+/// splits [str] into 2 from last occurence of [sep]
+List<String> cutFromLast(String str, String sep) {
+  final li = str.lastIndexOf(sep);
+  if (li == -1) {
+    return ['', str];
+  }
+  return [str.substring(0, li), str.substring(li + 1)];
+}
+
+String getLastName(String binaryName) => binaryName.split('.').last;
+
+String getSimpleNameOf(ClassDecl cls) => cls.simpleName;
+
+/// Returns class name as useful in dart.
+/// Eg -> a.b.X.Y -> X_Y
```

Comment:
new line in between single line summary and example.

https://dart.dev/guides/language/effective-dart/documentation#do-start-doc-comments-with-a-single-sentence-summary

Reply:
oops I knew this. But it's taking me time to internalize these rules haha.